core/app/models/spree/line_item.rb

```diff
@@ -147,6 +149,15 @@ def options=(options = {})
       assign_attributes opts
     end
 
+    def from_promotion
```

Comment:
This is useful when we want to display proper info on eg. the storefront - instead of price, quantity picker, etc.